src/layout/rect.rs

```diff
@@ -86,6 +116,75 @@ impl Rect {
         }
     }
 
+    /// Adjusts the size of the `Rect`.
+    ///
+    /// Adjust the size of the `Rect` by moving either the top or left. This effectively changes
+    /// [`Rect::x`] and [`Rect::y`] and recalculate [`Rect::width`] and [`Rect::height`] so that
+    /// the right and bottom "borders" do not move.
+    ///
+    /// - Positive `x` moves the left "border" to the right, negative to the left.
+    /// - Positive `y` moves the top "border" to the bottom, negative to the top.
+    ///
+    /// # Caveats
+    ///
+    /// If the given offset overflows the `Rect`'s fields, resulting values will wrapped around.
```

Comment:
The idea was: if `offset.x + rect.x` was greater than `u16::MAX` it would go back to 0.

Concrete example:
```rust
// NOTE u16::MAX = 65535
assert_eq(65000_u16 + 2000_u16, 1465_u16)
```

---

Anyway you're right that's not really useful, especially going towards 0 you wouldn't want to wrap around to 65 thousands ...